spec/factories/rankinglists.rb

```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :rankinglist do
```

Comment:
make this a child of event by using `parent: :event` and set the class to rankinglist by `class: :rankinglist` See the tournament and league factory on dev if you want an example. Also if you end up removing the start and enddate from the even factory itself (because it isn't needed for rankinglists) please add it into the tournament and league factories.

Reply:
I dont think it is necessary to remove ':deadline' and ':event' from the events factory, since you could also just set those values to nil in the ranking list values when they don't get validated.

Reply:
Factories are supposed to create the minimal needed version of a model though, so if a rankinglist doesn't need these dates we should remove it from the model as necessary validation and therefore remove it from the factory aswell.
We could however add a subfactory :event_with_dates which has these values and then let :league and :tournament factories inherit from this factory instead of the base one. That way we don't have to set the values to nil in the rankinglist factory.

Reply:
Well, if the dates are required in the events model, but not in the ranking list model, then setting them in the event factory and setting them to nil in the ranking factory would be the best option in my opinion. That way there are only required values initialized by the factory and the others are nil.

If the validations get moved to the submodels, then I agree, the factories should also display that movement.